Chap_API_Server.tex

```diff
@@ -1308,6 +1311,104 @@ \subsubsection{Initialize the cpuset structure}
 \end{arglist}
 
 
+%%%%%%%%%%%
+\subsection{\code{PMIx_server_define_process_set}}
+\declareapi{PMIx_server_define_process_set}
+
+%%%%
+\summary
+
+Add members to a \ac{PMIx} process set
+
+%%%%
+\format
+
+\versionMarker{4.0}
+\cspecificstart
+\begin{codepar}
+pmix_status_t
+PMIx_server_define_process_set(const pmix_proc_t *members,
+                               char *pset_name);
+\end{codepar}
+\cspecificend
+
+\begin{arglist}
+\argin{members}{Pointer to an array of \refstruct{pmix_proc_t} containing the
+identifiers of the processes to be added to the process set (handle)}
+\argin{pset_name}{String name of the process set to which the specified
+processes are to be added (\code{char*})}
+\end{arglist}
+
+Returns either \refconst{PMIX_SUCCESS} or an appropriate \ac{PMIx} error constant.
+
+
+%%%%
+\descr
+
+Provide a function by which the host environment can add processes to a
+process set, creating the set if it doesn't already exist. The \ac{PMIx}
+server shall alert all local clients of the new process set (including process
+set name and membership) via the \refconst{PMIX_PROCESS_SET_DEFINE} event.
+
+\advicermstart
+The host environment is responsible for ensuring:
+
+\begin{itemize}
+    \item consistent knowledge of process set membership across all involved
+    \ac{PMIx} servers; and
+    \item that process set names do not conflict with system-assigned
+    namespaces within the scope of the set
+\end{itemize}
+
+\advicermend
+
+%%%%%%%%%%%
+\subsection{\code{PMIx_server_delete_process_set}}
+\declareapi{PMIx_server_delete_process_set}
+
+%%%%
+\summary
+
+Delete members from a \ac{PMIx} process set
+
+%%%%
+\format
+
+\versionMarker{4.0}
+\cspecificstart
+\begin{codepar}
+pmix_status_t
+PMIx_server_delete_process_set(const pmix_proc_t *members,
+                               char *pset_name);
+\end{codepar}
+\cspecificend
+
+\begin{arglist}
+\argin{members}{Pointer to an array of \refstruct{pmix_proc_t} containing the
+identifiers of the processes to be removed from the process set (handle)}
+\argin{pset_name}{String name of the process set from which the specified
+processes are to be removed (\code{char*})}
+\end{arglist}
+
+Returns either \refconst{PMIX_SUCCESS} or an appropriate \ac{PMIx} error
+constant.
+
+
+%%%%
+\descr
+
+Provide a function by which the host environment can remove processes from a
+process set. If the array of members is \code{NULL}, then all current members
+are to be removed and the process set name deleted from the \ac{PMIx} server.
+The \ac{PMIx} server shall alert all local clients of the process set
+modifications via the \refconst{PMIX_PROCESS_SET_DELETE} event.
+
```

Comment:
I think we should put in a line about empty sets since we have the NULL clarification. First, I guess should we allow for the existence of a set with no members? It might be a useful concept, but I don't have a use case for it. @hppritcha Does the MPI sessions proposal allow for empty psets?

If the set has 1 process and the server specifies that one process in the `members` list instead of `NULL` does the set get deleted just like if they would have passed `NULL`? Or does the server have to pass `NULL` to delete the set?

Reply:
Ah - per section 13.1.1 "if all members of the set have been removed, then the set definition has been deleted from the system". I would just add that sentence here for clarity.

Reply:
Updated in latest commit